### ISSUE TITLE
Enhancing excluded_models with different schemas

### ISSUE THREAD

**Issue by cmager** [influitive/apartment#268]
hi, I needed to have different schemas for excluded_models, specificly for PostgreSQL in my case so i have modified `PostgresqlSchemaAdapter#process_excluded_models`, starting line 66 of https://github.com/roompad/apartment/blob/feature/postgresql-extend-process-excluded-models/lib/apartment/adapters/postgresql_adapter.rb

Now I can configure excluded_models like this (compatible to the string array config):

```ruby
Apartment.configure do |config|
  config.excluded_models = {
    public: %w(Tenat User),
    store: %w(Order Products)
  }
end
```

That is resulting in the following table_names:

> Tenant.table_name
> => 'public.tenants'
>
> Order.table_name
> => 'store.orders'
 
This is working very well for me. Is this interesting for to open a pull request so others can adopt this for other adapters?

**Comment by bradrobertson** [comment]
I've actually wanted to remove the `public` constraint from `excluded_models` and make it completely configurable (but default to `public` if people don't care). So I think this makes sense, BUT I don't want a conditional check in the adapter itself. The reason you have a condition is because you've been passed a heterogeneous object (`Hash` or `Array`). What if it was always a hash?

So, if someone just passes an array, you'd construct a hash of `{public: %w(Tenant User)}` in the configuration, then the adapter itself doesn't need to have any guards, it always just assumes to map over a hash

So basically the config interface *could* be:

```ruby
Apartment.configure do |config|
  config.excluded_models = %w(Tenant User)
end
```
which would store it as `{public: %w(Tenant User)}`

Or you could specifically pass it the map of schema name to tenant_name as you mentioned above.

**Comment by bradrobertson** [comment]
Note also that this applies to the `mysql2` adapter

**Comment by alepore** [comment]
@cmager thanks, i have the same need and i'm using your tweak.
would love to have this integrated into apartment!
